### PR TITLE
SCT-3129: Improve GHA workflow

### PR DESCRIPTION
**Pull request by eapearson**
This set of changes purports to improve the GitHub action workflow in several respects:

- unifies into a single test, build, push workflow
- triggered by pushes to main, feature, and fix branches
- triggered by pull request activity
- triggered by a release
- pushes to GHCR rather than Dockerhub

The motivation for these changes is that I'm working on a set of improvements which I need to deploy for review before finalizing. This is a general need - to be able to iterate on a feature branch, including making it available for review by interested parties, before formalizing in a Pull Request. This is of special importance to UI projects, as they are tangible, usable instruments which, from experience, most reviewers or interested parties will not install locally in order to try them out.

The change to GHCR from Dockerhub is just because the is our standard practice now, and any reworking of GHA support should follow current practices.

Some topics for specific review and discussion:

## Triggering Branches

I've followed the practice I had instituted in kbase-ui to trigger builds only for specific branches or branch patterns, in this case `main`, a "feature branch" identified prefixed by `feature-` and a "fix branch" prefixed by `fix-`. The first branch is not controversial, but perhaps the latter two merit some discussion.

But why limit test and build to just these types of branches? If most branches are either main, feature, or fix branches, they why not just test and build for all branches? If an odd branch here or there is created and not intended for deployment anywhere, does it hurt to catch them in the GHA workflow? They will benefit from being tested, and knowing that an image can be built.

Similarly, why limit the branch naming thus? I've found it is helpful in kbase-ui, but part of that is because it helps sift through the cruft of old branches (which should just be deleted.) This is a relatively new repo, and does not suffer this problem. What if, instead, we always use the Jira ticket id as the prefix for non-main branches? Then the branch is tied to the ticket information and activity. After all, distinction between feature and fix is sometimes fuzzy anyway.

We could even enforce that in the workflow, if we limit test and build only branches beginning with, e.g., "PTV-" and "SCT-".

## Image Tag Name

The image tag name is formed by using the `CI_REF_NAME_SLUG` environments variable created by the `FranzDiebold/github-env-vars-action@v2` GitHub Action (https://github.com/FranzDiebold/github-env-vars-action). This action is recommended by the GHA team, and serves to create additional useful environment variables that could have been, but were not, supplied by the native GHA workflow support from GitHub. This particular env variable is a "slugified" version of the branch name, tag name or PR number that triggered the workflow.

The slugification process converts troublesome characters to "-" and ensures that the value can be used as an image tag, and for other purposes. For our purposes, the only notable outcome is that pull requests have the tag format "#-merge", where # is the PR number, and releases have the format "v#-#-#" if the tag is a semantic version tag like "v#.#.#". It seems like a reasonable approach, but it is different that might be naively expected.

### CODE REVIEW

File: .github/workflows/test-and-build.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+---
+name: Build and Push Up Feature or Fix Branch Image
+on:
```
Comment (dakotablair) [inline .github/workflows/test-and-build.yml:3]: If you would, please add `workflow_dispatch` as a trigger as well. Granted, it's easy enough to make a branch named `fix-*` to build from, but having `workflow_dispatch` is really useful for updating actions.

**Review by dakotablair — CHANGES_REQUESTED** [review]
This looks good to me. I commented on a few typos, and also request that you add `workflow_dispatch` as a trigger for the test and build action for future use.

File: docs/github-actions.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+## Release `v#-#-#`
+
+A release created through the GitHub release tool will trigger the workflow and create an image with a tag like `v#-#-#`. This tag is very similar to the git tag created during a release, which is based on a semantic version, and looks like `v#.#.#`.
+
+Since the tag is based on `CI_REF_NAME_SLUG`, the "slugification" process converts the `.` to `-` in order to make it more universally acceptable as an identifier.
+
+A release image is based on the state of the `main` branch at some point at which the team decides a new release is suitable. A release image is suitable for deployment in any user-facing environment, including `appdev` and `prod` (aka `narrative.kbase.us`). It may also be deployed to `next`, which serves as a pre-production evalutation environment.
```
[the remark this answers is not in the document]
Comment (eapearson) [reply docs/github-actions.md:75]: :)

[the remark this answers is not in the document]
Comment (eapearson) [reply docs/github-actions.md]: Oops, I recently installed my IDE and had not yet installed the spellchecker, so of course all my spelling was always perfect, no typoos.
I've installed it, and a code spell checker, and fixed my typos and a few existing ones in the codebase, including one which was a misspelling of a user profile setting which was a latent, if innocuous bug (it prevented the user setting which disabled gravatar from working, which also revealed a bug in the user profile app, which is a bug for another day...)

File: docs/github-actions.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Github Actions
+
+Certain events in the Navigator GitHub reopo trigger a GitHub Action (GHA) workflow which runs tests, builds an image, and push the image to GitHub Container Registry (GHCR). A single GHA workflow located in `.github/workflows/test-and-build.yml` is invoked for all actions. This workflow will run unit tests, and, should they succeed, build an image and push it up to the GitHub Container Registry (GHCR).
```
[the remark this answers is not in the document]
Comment (eapearson) [reply docs/github-actions.md]: fixed (and a few other typos as well)

**Pull request merged**